hda-install

```diff
@@ -1300,27 +1308,37 @@ def main
 	end
 
 	usage if @opt["u"]
+
+        @system_configuration = nil
         
 	do_basic_checks
 
 	set_standard_out
 
         #to support both INSTALL CODE or no INSTALL CODE
-        if ARGV.size > 0 && ARGV[0].length > 4
+        if (@opt["k"] && ARGV.size == 2 && ARGV[0].length > 4) || (!@opt["k"] && ARGV.size > 0 && ARGV[0].length > 4) #if -k option there, then network interface is always given. So, no of args, if not 2, then INSTALL CODE is not used
```

Comment:
This look like a source of future trouble to me.

All these options need to be readable and understandable.

Reply:
Well, I have tried to test them, with every possible outcome. 
Added comments in detail.

Reply:
Maybe we need another option to install using INSTALL CODE.
The thing is, now, for the first time, both `-k` option and `INSTALL CODE` requires argument.
And `INSTALL CODE` is provided without any option. So, no way of knowing that whether the argument is `INSTALL CODE` or 'network Interface`.